pkg/kbuser/object_ids.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kbuser
+
+import (
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AnalyzeObjectID examines an object in a kube-bind service provider
+// cluster and returns the object's ID as known in the service consumer
+// cluster and the kube-bind ID for the consumer, or an error if
+// the object does not appear to be a provider's copy of a consumer's object.
+func AnalyzeObjectID(obj metav1.Object) (namespace, name, kbSpaceID string, err error) {
+	namespace = obj.GetNamespace()
+	name = obj.GetName()
+	if namespace == "" {
+		name, kbSpaceID, err = parseMappedName(name)
+		return
+	}
+	namespace, kbSpaceID, err = parseMappedName(name)
```

Comment:
Is there a case for name spaced object ?

Reply:
I am not sure what we have in the future. I thought it best to not bet against caring about namespaced objects.